### DISCUSSION TITLE
Async image processing with Django Celery not working

### DISCUSSION THREAD

**Discussion by soramikan** [codingjoe/django-pictures#158] [Q&A]
The task is loaded, but when I sent the image through the admin site, nothing happened.
Another task I implemented will be executed without problems, and there seems to be no problem with Celery.

```
 -------------- celery@*** v5.3.6 (emerald-rush)
--- ***** ----- 
-- ******* ---- macOS-14.0-arm64-arm-64bit 2024-01-30 13:31:43
- *** --- * --- 
- ** ---------- [config]
- ** ---------- .> app:         ***:**********
- ** ---------- .> transport:   redis://localhost:6379/0
- ** ---------- .> results:     
- *** --- * --- .> concurrency: 8 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** ----- 
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery
                

[tasks]
  . pictures.tasks.process_picture_with_celery

[2024-01-30 13:31:44,321: INFO/MainProcess] Connected to redis://localhost:6379/0
[2024-01-30 13:31:44,324: INFO/MainProcess] mingle: searching for neighbors
[2024-01-30 13:31:45,349: INFO/MainProcess] mingle: all alone
[2024-01-30 13:31:45,367: INFO/MainProcess] celery@*** ready.
```

**Comment by soramikan** [comment]
I found a way to temporarily solve the problem by not doing asynchronous processing **temporarily**, so I will share it.

Make the following settings in `settings.py`.
```python
PICTURES = {
    "BREAKPOINTS": {
        "xs": 576,
        "s": 768,
        "m": 992,
        "l": 1200,
        "xl": 1400,
    },
    "GRID_COLUMNS": 12,
    "CONTAINER_WIDTH": 1200,
    "FILE_TYPES": ["WEBP"],
    "PIXEL_DENSITIES": [1, 2],
    "USE_PLACEHOLDERS": False,
    "QUEUE_NAME": "pictures",
    "PROCESSOR": "pictures.tasks._process_picture",    # Change here !
}
```
Change `PROCESSOR` to `pictures.tasks._process_picture`.
Note that `_` is at the beginning of `process_picture`.

**Comment by codingjoe** [comment]
Hi there 👋,

Thanks for reaching out. In your second comment, you simply switched to synchronous processing. Which will work, but I wouldn't recommend.

I believe the solution is simple and – maybe not well enough – documented:

> You will need workers to listen to the `pictures` queue.

If you need any assistance, please don't hesitate to reach out here on the Celery community.

Cheers!
Joe
